File: IT/Eccezioni/Python/exceptionPropagation.py
# file exceptionPropagation.py 

from __future__ import print_function

# la funzione propagate() solleva un'eccezione generica 

def propagate(): 
	raise Exception("Sollevata un'eccezione!") 

# la funzione chiamante caller() invoca la funzione propagate() e gestisce l'eccezione

def caller(): 
	try: 
		propagate() 

	except Exception as e: 
		print(str(e))

# invoca la funzione caller() 

caller() 

print("Fine del programma.")

File: IT/Eccezioni/Java/ExceptionPropagation.java
/* file ExceptionPropagation.java */

public class ExceptionPropagation {
	/* il metodo statico propagate() solleva un'eccezione generica, propagata all'esterno con la parola chiave throws */

	static void propagate() throws Exception {
		throw new Exception("Sollevata un'eccezione!");
	}

	/* il metodo main() invoca il metodo propagate() e gestisce l'eccezione */

	public static void main(String [] args) {
		try {
			propagate();
		} 
		catch (Exception e) {
			e.printStackTrace();
		}

		System.out.println("Fine del programma.");
	}
}

